hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -64,6 +64,14 @@ public static String getNullableNestedFieldValAsString(GenericRecord record, Str
     }
   }
 
+  public static Object getNullableNestedFieldVal(GenericRecord record, String fieldName) {
+    try {
+      return getNestedFieldVal(record, fieldName);
```

Comment:
Instead of letting getNestedFieldVal throw an exception and then handling it (more compute intensive), can you add pass additional flag to getNestedFieldVal to return null instead of throwing the exception ?

Reply:
Point well taken. Will do the changes.

Reply:
@bvaradar Actually I wrote the above function following the way getNullableNestedFieldValAsString function is written in the same class. Should I change the usages of that function also by passing a flag like how you are suggesting? That function is also compute intensive.

Reply:
@pratyakshsharma : Yeah, the other function also needs to be fixed. Thank you.

Reply:
Done with the changes. @bvaradar